File: client-java/instrumentation/src/test/java/com/foo/somedifferentpackage/examples/methodreplacement/strings/StringsExampleImp.java
package com.foo.somedifferentpackage.examples.methodreplacement.strings;

import org.evomaster.client.java.instrumentation.coverage.methodreplacement.classes.StringClassReplacement;
import org.evomaster.client.java.instrumentation.example.methodreplacement.strings.StringsExample;

import static org.evomaster.client.java.instrumentation.shared.ObjectiveNaming.METHOD_REPLACEMENT;

public class StringsExampleImp implements StringsExample {

    @Override
    public boolean isFooWithDirectReturn(String value) {
        return "foo".equals(value);
    }

    @Override
    public boolean isFooWithDirectReturnUsingReplacement(String value){
        return new StringClassReplacement().equals("foo", value, METHOD_REPLACEMENT);
    }

    @Override
    public boolean isFooWithBooleanCheck(String value) {
        return "foo".equals(value) == true;
    }

    @Override
    public boolean isFooWithNegatedBooleanCheck(String value) {
        return "foo".equals(value) != false;
    }

    @Override
    public boolean isFooWithIf(String value) {
        if("foo".equals(value)){
            return true;
        } else {
            return false;
        }
    }

    @Override
    public boolean isFooWithLocalVariable(String value) {

        boolean local = "foo".equals(value);

        return local;
    }

    @Override
    public boolean isFooWithLocalVariableInIf(String value) {

        boolean local;
        if("foo".equals(value)){
            local = true;
        } else {
            local = false;
        }

        return local;
    }

    @Override
    public boolean isNotFooWithLocalVariable(String value) {

        boolean local = ! "foo".equals(value);

        return local;
    }

    @Override
    public boolean isBarWithPositiveX(String value, int x) {

        boolean local = value.equals("bar") && x > 0;

        return local;
    }

}

File: outdated/outdated-client-dotnet/test/EvoMaster.Instrumentation.Examples/Strings/StringsExampleImp.cs
namespace EvoMaster.Instrumentation.Examples.Strings{
    public class StringsExampleImp : IStringsExample{
        
        public bool isFooWithDirectReturn(string value) {
            return "foo".Equals(value);
        }

        // public bool isFooWithDirectReturnUsingReplacement(string value){
        //     return StringClassReplacement.Equals("foo", value, METHOD_REPLACEMENT);
        // }

        public bool isFooWithBooleanCheck(string value) {
            return "foo".Equals(value) == true;
        }

        public bool isFooWithNegatedBooleanCheck(string value) {
            return "foo".Equals(value) != false;
        }

        public bool isFooWithIf(string value) {
            if("foo".Equals(value)){
                return true;
            } else {
                return false;
            }
        }

        public bool isFooWithLocalVariable(string value) {

            bool local = "foo".Equals(value);

            return local;
        }

        public bool isFooWithLocalVariableInIf(string value) {

            bool local;
            if("foo".Equals(value)){
                local = true;
            } else {
                local = false;
            }

            return local;
        }

        public bool isNotFooWithLocalVariable(string value) {

            bool local = ! "foo".Equals(value);

            return local;
        }

        public bool isBarWithPositiveX(string value, int x) {

            bool local = value.Equals("bar") && x > 0;

            return local;
        }
    }
}